src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -238,12 +238,26 @@ def check_cluster_outbound_connectivity(outbound_connectivity_check_log, filepat
         outbound_connectivity_response = outbound_connectivity_response[::-1]
         if consts.Outbound_Connectivity_Check_Result_String not in outbound_connectivity_check_log:
             return consts.Diagnostic_Check_Incomplete, storage_space_available
+
+        # extracting the obo endpoint log specifically from outbound connectivity check job logs
+        obo_check_log = outbound_connectivity_check_log.split('  ')[0]
+        # extracting the obo endpoint url
+        obo_enpoint_url = obo_check_log.split(" : ")[1]
+        # extracting the obo endpoint response code from outbound connectivity check
+        obo_check_response_code = obo_check_log.split(" : ")[2]
         # Validating if outbound connectiivty is working or not and displaying proper result
         if(outbound_connectivity_response != "000"):
+            # since connectivity to obo endpoint is required only for cluster connect feature , we only give out a warning if connectivity to obo endpoint is not present
+            if obo_check_response_code == "000":
```

Comment:
Can't we have this condition outside of the parent if check? That way we keep it independent and will be easy to increase the list of endpoints in each category easily in future

Reply:
The reason for keeping this inside the if condition was , that supposing that the outbound connectivity isnt there to any endpoint , it will still give out the warning about obo endpoint and then it will say outbound connectivity fail when it checks the mcr endpoint.

So i thought if mcr itself fails , then pointing to obo specifically doesnt make sense and the user should look into all the endpoint network requirements document. And if mcr connectivity is there than pointing out if some specific endpoint is not listed makes more sense.